### ISSUE TITLE
Issue to train translation task on mutltiple GPU

### ISSUE THREAD

**Issue by pras06** [pytorch/fairseq#1466]
Hi,

I am trying to train a model to have a translation FR2EN on 4 GPU machine. I followed translation guidlines given [HERE](https://fairseq.readthedocs.io/en/latest/getting_started.html#training-a-new-model) .

Here is my command line that I used to train:
`fairseq-train data-bin/wmt14_en_fr --lr 0.25 --clip-norm 0.1 --dropout 0.2 --max-tokens 1500 --arch fconv_wmt_en_fr --save-dir checkpoints/fconv`

Here the error that I have:
![g3 16xlarge_error1](https://user-images.githubusercontent.com/57360151/70290224-7bc2e880-17d7-11ea-85f7-b31a16471165.PNG)

As recommended in some your posts I have adjusted my batch size to avoid memory issue, here the memory usage of my machine:
![g3 16xlarge_GPUusage](https://user-images.githubusercontent.com/57360151/70290280-af057780-17d7-11ea-97bd-effee3c70813.PNG)

Could you please give me some advices about have to solve it please? I am using AWS machines, it's costing some money, it would be great if you can help me :)

Or maybe somebody here can give a model of translation doing FR2EN, I know have the model for EN2FR, but I didn't find the model for FR2EN, that's why I am training this model.

Thanks,
Pras

**Comment by myleott** [comment]
This looks like there’s something wrong with your numpy setup. Can you upgrade numpy and rebuild the fairseq Cython modules?

**Comment by pras06** [comment]
Hi,

Thanks for your help, yes it was an issue with numpy version, initially I cloned an existing env instead of creating a new one. Now it's fine, it's training well, so you can close this issue. 

Just a question about the training, how many epochs are defined by default? Where can I find this information? How or where can I change that?

Thanks a lot
Pras

**Comment by myleott** [comment]
Great! The termination conditions are listed here: https://github.com/pytorch/fairseq/blob/master/train.py#L73-L84

**Comment by pras06** [comment]
Great, thanks a lot :)
